There is one failing test but that failed before already (unless I am super confused) so it's not my department:
```
ERROR: testCreateFieldEqualityExpression (__main__.TestQgsExpressionCustomFunctions)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/hannes/.cache/yay/qgis-git/src/qgis/tests/src/python/test_qgsexpression.py", line 283, in testCreateFieldEqualityExpression
    self.assertEqual(e.createFieldEqualityExpression(field, value, type), res)
TypeError: QgsExpression.createFieldEqualityExpression(): too many arguments
```

--- a/tests/src/python/test_qgsexpression.py
+++ b/tests/src/python/test_qgsexpression.py
@@ -294,11 +294,11 @@ class TestQgsExpressionCustomFunctions(unittest.TestCase):
         e.setExpression("foo = 1")
         self.assertTrue(e.isValid())
         self.assertEqual(len(e.referencedAttributeIndexes(QgsFields())), 0)
-        
+
     def testSuccessfulEvaluationReturnsNoEvalErrorString(self):
         exp = QgsExpression("True is False")  # the result does not matter
         self.assertEqual(exp.evalErrorString(), "")
-        
+
     def testExceptionDuringEvalReturnsTraceback(self):
         QgsExpression.registerFunction(self.raise_exception)
         exp = QgsExpression('raise_exception()')

--- a/tests/src/python/test_qgsexpression.py
+++ b/tests/src/python/test_qgsexpression.py
@@ -310,7 +310,7 @@ class TestQgsExpressionCustomFunctions(unittest.TestCase):
             "  File \".*qgsfunction.py\", line [0-9]+, in func\n"
             "    return self.function\\(\\*values\\)\n"
             "  File \".*test_qgsexpression.py\", line [0-9]+, in raise_exception\n"
-            "    foo  # an undefined variable\n"
+            "    foo  # noqa: F821\n"
             "NameError: name \'foo\' is not defined"
             "\n</pre>"
         )

--- a/tests/src/python/test_qgsexpression.py
+++ b/tests/src/python/test_qgsexpression.py
@@ -72,7 +72,8 @@ class TestQgsExpressionCustomFunctions(unittest.TestCase):
 
     @qgsfunction(group='testing', register=False)
     def raise_exception(feature, parent):
-        foo  # an undefined variable
+        # an undefined variable
+        foo  # noqa: F821
 
     def tearDown(self):
         QgsExpression.unregisterFunction('testfun')

